sphinxext/opengraph/__init__.py

```diff
@@ -38,10 +40,17 @@ def get_tags(
     doctree: nodes.document,
     config: Dict[str, Any],
 ) -> str:
+    # Get field lists for per-poge overrides
+    fields = context["meta"]
+    if fields is None:
+        fields = {}
 
     # Set length of description
     try:
-        desc_len = int(config["ogp_description_length"])
+        try:
+            desc_len = int(fields["ogp-description-length"])
+        except (ValueError, KeyError):
+            desc_len = int(config["ogp_description_length"])
     except ValueError:
         desc_len = DEFAULT_DESCRIPTION_LENGTH
```

Comment:
This would be more readable as 1 try with 2 excepts.

Reply:
Not sure I understand what you mean by 1 try with 2 excepts? Although it might go down to 1 try when I use `fields.get` anyways.. (Need to check if that works)

Reply:
You can chain excepts like elifs

Reply:
But a `ValueError` can be thrown inside both the inner try and the except, specifically, if they were chained excepts, and the config contained a non-integer value, a `ValueError` would be thrown.